cardano-cli/src/Cardano/CLI/Read.hs

```diff
@@ -519,6 +522,22 @@ data ShelleyBootstrapWitnessSigningKeyData
       -- If specified, both the network ID and derivation path are extracted
       -- from the address and used in the construction of the Byron witness.
 
+-- | Construct a Shelley bootstrap witness (i.e. a Byron key witness in the
+-- Shelley era).
+mkShelleyBootstrapWitness
+  :: ()
+  => ShelleyBasedEra era
+  -> Maybe NetworkId
+  -> A.TxBody era
```

Comment:
:man_facepalming: I'm just noticing this type. What do we can by wrapping the ledger type here? Why not directly use the ledger type?

Reply:
That we don't have to use `ShelleyLedgerEra` type family and we can hide some calls to `shelleyBasedEraConstraints` behind our lens instead of ledger lens. A little convenience for some boilerplate.

If we decide to not use this `TxBody` type, we should deprecate the module `Cardano.Api.Ledger.Lens` then.